spec/suites/layer/GeoJSONSpec.js

```diff
@@ -512,3 +512,387 @@ describe("L.LayerGroup#toGeoJSON", function () {
+			expect(latlng.alt).to.be(0);
+		});
+	});
+
+	describe("#coordsToLatLngs", function () {
+
+		function customCoordsToLatLng(coords) {
+			return new L.LatLng(coords[0] + 1, coords[1] + 1, coords[2] + 1);
+		}
+
+		it("creates a multidimensional array of LatLngs", function () {
+			const latLngs = L.GeoJSON.coordsToLatLngs([[0, 0], [1, 1], [2, 2]]);
+			expect(latLngs.length).to.be(3);
+			latLngs.forEach(function (latLng) {
+				expect(latLng instanceof L.LatLng).to.be(true);
+			});
+		});
+
+		it("creates a multidimensional array of LatLngs (levelsDeep=1)", function () {
+			const latLngs = L.GeoJSON.coordsToLatLngs([
+				[[0, 0], [1, 1], [2, 2]],
+				[[4, 4], [5, 5], [6, 6]]
+			], 1);
+			expect(latLngs.length).to.be(2);
+			for (var i = 0; i < latLngs.length; i++) {
+				for (var j = 0; j < latLngs[i].length; j++) {
+					expect(latLngs[i][j] instanceof L.LatLng).to.be(true);
+				}
+			}
+		});
+
+		it("creates a multidimensional array of LatLngs with custom coordsToLatLng", function () {
+			const coords = [[1, 2, 3], [4, 5, 6], [7, 8, 9]];
+			const latLngs = L.GeoJSON.coordsToLatLngs(coords, 0, customCoordsToLatLng);
+			const expected = [
+				{lat: 2, lng: 3, alt: 4},
+				{lat: 5, lng: 6, alt: 7},
+				{lat: 8, lng: 9, alt: 10}
+			];
+			for (var i = 0; i < latLngs.length; i++) {
+				expect(latLngs[i]).to.eql(expected[i]);
+			}
+		});
+
+		it("creates a multidimensional array of LatLngs with custom coordsToLatLng (levelDeep=1)", function () {
+			const coords = [
+				[[1, 2, 3], [4, 5, 6]],
+				[[12, 13, 14], [15, 16, 17]]
+			];
+			const latLngs = L.GeoJSON.coordsToLatLngs(coords, 1, customCoordsToLatLng);
+			const expected = [
+				[
+					{lat: 2, lng: 3, alt: 4},
+					{lat: 5, lng: 6, alt: 7}
+				],
+				[
+					{lat: 13, lng: 14, alt: 15},
+					{lat: 16, lng: 17, alt: 18}
+				]
+			];
+			for (var i = 0; i < latLngs.length; i++) {
+				for (var j = 0; j < latLngs[i].length; j++) {
+					expect(latLngs[i][j]).to.eql(expected[i][j]);
+				}
+			}
+		});
+	});
+
+	describe("#latLngToCoords", function () {
+		it("returns an array of coordinates (longitude, latitude)", function () {
+			const coords = L.GeoJSON.latLngToCoords(new L.LatLng(0, 0));
+			expect(coords).to.eql([0, 0]);
+		});
+
+		it("returns an array of coordinates and altitude (longitude, latitude, altitude)", function () {
+			const coords = L.GeoJSON.latLngToCoords(new L.LatLng(0, 0, 0));
+			expect(coords).to.eql([0, 0, 0]);
+		});
+
+		it("returns an array of coordinates with given precision (longitude, latitude)", function () {
+			const coords = L.GeoJSON.latLngToCoords(new L.LatLng(
+				1.123456, 1.123456
+			), 3);
+			expect(coords).to.eql([1.123, 1.123]);
+		});
+
+		it("returns an array of coordinates with given precision (longitude, latitude, altitude)", function () {
+			const coords = L.GeoJSON.latLngToCoords(new L.LatLng(
+				1.123456, 1.123456, 1.123456
+			), 3);
+			expect(coords).to.eql([1.123, 1.123, 1.123]);
+		});
+	});
```

Comment:
Better combine these two tests into one.

(Perhaps there are more such tests that should be combined)

Reply:
But why combine them? They are two different cases.

Reply:
Well, I agree that this is rather matter of taste.
As for this case, I feel that `[lat, lng]` is kind of special case of general `[lat, lng, alt]`.
Sure, there is nothing wrong to put it into separate test, but see:
- having too many discrete tests can be a bit confusing
- we do not lose anything if we put two similar tests together, as when test fails - we see where it fails (exact line).

Reply:
> As for this case, I feel that `[lat, lng]` is kind of special case of general `[lat, lng, alt]`.

I think its more the general case to only have `[lat, lng]`, as until now i have never had any GeoJSON dataset that also had altitude values.  

> having too many discrete tests can be a bit confusing

Well, in this case the test description clearly tells what is tested, so i don't see any confusion here to be honest :)

> we do not lose anything if we put two similar tests together, as when test fails - we see where it fails (exact line).

Sure you are right, but with two different cases you instantly know which of the test failed by description. You wouldn't have to read line numbers. But yeah, you are right this is a matter of taste. Personally i prefer to put such cases into seperate tests, as for me DRY doesn't necessarily count in tests. I will put those two tests into one.

Reply:
> you instantly know which of the test failed by description. You wouldn't have to read line numbers

But to fix broken case you still need to inspect exact line.
Pay attention, some other your tests already have several `expect` checks (btw, test may fail not only because of `expect`).

Reply:
> btw, test may fail not only because of expect 

huh? what do you mean exactly with that? programming errors? just curious...

Reply:
`expect` is just special way to throw exception when some expectation not met.
Obviously that if any part of code throws when it shouldn't - test also fails.

Reply:
ok yeah thats true... test code is just code that can have logical or syntax errors